src/neo/Oracle/OraclePolicyContract.cs

```diff
@@ -0,0 +1,167 @@
+            UInt160 account = new UInt160(args[0].GetSpan());
+            if (!InteropService.Runtime.CheckWitnessInternal(engine, account)) return false;
+            ECPoint[] pubkeys = ((Array)args[1]).Select(p => p.GetSpan().AsSerializable<ECPoint>()).ToArray();
+            if (pubkeys.Length != 2) return false;
+            StoreView snapshot = engine.Snapshot;
+            StorageKey key = CreateStorageKey(Prefix_Validator, pubkeys[0]);
+            if (snapshot.Storages.TryGet(key) != null)
+            {
+                StorageItem value = snapshot.Storages.GetAndChange(key);
+                value.Value = pubkeys[1].ToArray();
+            }
+            else
+            {
+                snapshot.Storages.Add(key, new StorageItem
+                {
+                    Value = pubkeys[1].ToArray()
+                });
+            }
+            return true;
+        }
+
+        [ContractMethod(0_03000000, ContractParameterType.Array)]
+        private StackItem GetOracleValidators(ApplicationEngine engine, Array args)
+        {
+            return new Array(engine.ReferenceCounter, GetOracleValidators(engine.Snapshot).Select(p => (StackItem)p.ToArray()));
+        }
+
+        public ECPoint[] GetOracleValidators(StoreView snapshot)
+        {
+            ECPoint[] consensusPublicKey = PolicyContract.NEO.GetValidators(snapshot);
+            IEnumerable<(ECPoint ConsensusPublicKey, ECPoint OraclePublicKey)> hasDelegateOracleValidators = GetDelegateOracleValidators(snapshot).Where(p => consensusPublicKey.Contains(p.ConsensusPublicKey));
+            foreach (var item in hasDelegateOracleValidators)
+            {
+                var index = System.Array.IndexOf(consensusPublicKey, item.ConsensusPublicKey);
+                if (index >= 0) consensusPublicKey[index] = item.OraclePublicKey;
+            }
+            return consensusPublicKey;
+        }
+
+        [ContractMethod(0_03000000, ContractParameterType.Integer)]
+        private StackItem GetOracleValidatorsCount(ApplicationEngine engine, Array args)
+        {
+            return GetOracleValidatorsCount(engine.Snapshot);
+        }
+
+        public BigInteger GetOracleValidatorsCount(StoreView snapshot)
+        {
+            return GetOracleValidators(snapshot).Length;
+        }
+
+        internal IEnumerable<(ECPoint ConsensusPublicKey, ECPoint OraclePublicKey)> GetDelegateOracleValidators(StoreView snapshot)
+        {
+            byte[] prefix_key = StorageKey.CreateSearchPrefix(Id, new[] { Prefix_Validator });
+            return snapshot.Storages.Find(prefix_key).Select(p =>
+            (
+                p.Key.Key.AsSerializable<ECPoint>(1),
+                p.Value.Value.AsSerializable<ECPoint>(1)
+            ));
+        }
+
+        internal UInt160 GetOracleMultiSigAddress(ECPoint[] validators)
+        {
+            return Contract.CreateMultiSigRedeemScript(validators.Length - (validators.Length - 1) / 3, validators).ToScriptHash();
+        }
```

Comment:
It's better to change the argument type of this method, as we can use in building OracleTransaction.
```c#
public UInt160 GetOracleMultiSigAddress(StoreView snapshot)
```

Reply:
ok